### PR TITLE
Task/tbq 14 find the service

### PR DESCRIPTION
**Pull request by Sekator778**
add only tests

### CODE REVIEW

File: src/main/java/ua/shpp/eqbot/command/AddService.java [hunk 1]
```diff
@@ -50,13 +49,13 @@ public AddService(SendBotMessageService sendBotMessageService, ServiceRepository
         this.bundleLanguage = bundleLanguage;
     }
 
-    public void addService(ServiceDTO service) {
+    public ServiceEntity addService(ServiceDTO service) {
         ServiceEntity serviceEntity = new ServiceEntity();
         serviceEntity.setTelegramId(service.getTelegramId())
                 .setName(service.getName())
                 .setDescription(service.getDescription())
                 .setAvatar(service.getAvatar());
```
Comment (AVM82) [inline src/main/java/ua/shpp/eqbot/command/AddService.java:57]: Read about this [https://mapstruct.org/](https://mapstruct.org/)

File: src/main/java/ua/shpp/eqbot/ecxeption/ValidationFailedException.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+public class ValidationFailedException extends RuntimeException {
+
+    public ValidationFailedException(String message) {
+        super(message);
+    }
+
+    public ValidationFailedException(String message, Throwable cause) {
+        super(message, cause);
+    }
+}
```
Comment (AVM82) [inline src/main/java/ua/shpp/eqbot/ecxeption/ValidationFailedException.java:12]:
```suggestion
}

```

File: src/main/java/ua/shpp/eqbot/validation/UserValidateService.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+public interface UserValidateService {
+
+    /**
+     * Checks whether the new user can be created.
+     * @param name - name user
+     * @param phone - phone user
+     */
+    void checkUserCreation(String name, String phone);
+}
```
Comment (AVM82) [inline src/main/java/ua/shpp/eqbot/validation/UserValidateService.java:11]:
```suggestion
}

```

File: src/main/java/ua/shpp/eqbot/validation/UserValidateServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    private final UserRepository userRepository;
+
+    @Override
+    @Transactional(readOnly = true)
+    public void checkUserCreation(String firstName, String lastName) {
+        if (userRepository.existsByNameAndPhone(firstName, lastName)) {
+            throw new ValidationFailedException("Such person already exists");
+        }
+    }
+}
```
Comment (AVM82) [inline src/main/java/ua/shpp/eqbot/validation/UserValidateServiceImpl.java:24]:
```suggestion
}

```

File: src/main/resources/db/migration/V2_0__init.sql [hunk 5]
```diff
@@ -1,25 +1,30 @@
...
-    name            varchar(255),
-    phone           varchar(255),
-    time_created    timestamp
+    id           bigserial
+        primary key,
+    city         varchar(255),
+    created_time timestamp,
+    language     varchar(255),
+    name         varchar(255),
+    phone        varchar(255),
```
Comment (AVM82) [inline src/main/resources/db/migration/V2_0__init.sql:9]:
```suggestion
    phone        varchar(20),
```
I think 20 will be enough

File: src/main/java/ua/shpp/eqbot/utility/ConverterDTO.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package ua.shpp.eqbot.utility;
+
+import org.modelmapper.ModelMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ua.shpp.eqbot.dto.UserDto;
+import ua.shpp.eqbot.model.UserEntity;
+
+public class ConverterDTO {
+    private static final ModelMapper modeMODEL_MAPPERMapper = new ModelMapper();
```
Comment (AVM82) [inline src/main/java/ua/shpp/eqbot/utility/ConverterDTO.java]: We really need a name like modeMODEL_MAPPERMapper

Comment (Sekator778) [reply src/main/java/ua/shpp/eqbot/utility/ConverterDTO.java]: omg! sorry

File: src/main/java/ua/shpp/eqbot/command/AddService.java (hunk 1 above)
Comment (Sekator778) [reply src/main/java/ua/shpp/eqbot/command/AddService.java:57]: I did it please see the last commit

File: src/main/java/ua/shpp/eqbot/utility/UserMapper.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package ua.shpp.eqbot.utility;
```
Comment (AVM82) [inline src/main/java/ua/shpp/eqbot/utility/UserMapper.java]: rename package utility to mapper

Comment (Sekator778) [reply src/main/java/ua/shpp/eqbot/utility/UserMapper.java]: i did it

**Review by AVM82 — APPROVED** [review]

**Review by Kornilchenko — APPROVED** [review]

**Pull request merged**